File: pset1/cash/cash.c
#include<stdio.h>
#include<cs50.h>
#include<math.h>

int change_owed(void);


int main(void)
{
    int n = change_owed();
    
    int coins = 0;  //how many coins in return
    
    while (n > 0)
    {
        
        if (n >= 25)
        { 
            //check how many 25 cents coins
            coins = coins + (n / 25);
            n = n % 25;
        }
        else if (n < 25 && n >= 10 && n > 5)
        { 
            //check how many 10 cents coins
            coins = coins + (n / 10);
            n = n % 10;
        }
        else if (n < 25 && n < 10 && n > 1)
        { 
            //check how many 5 cents coins
            coins = coins + (n / 5);
            n = n % 5;
        }
        else
        { 
            //check how many 1 cents coins
            coins += n;
            n = n % 1;
        }
    }
    
    printf("%i \n", coins);
}



int change_owed(void)
{
    float c;
    int cents; 
    
    do
    {
        c = get_float("Changed owed:"); //var for getting the change
    }
    while (c < 0);
    
    cents = round(c * 100);
    
    return cents;
    
}

File: pset6/cash/cash.py
import cs50

def main():
    
    n = change_owed()
    coins = 0
    
    while n > 0:
        if n >= 25:
            coins = coins + int(n / 25)
            n = n % 25
        elif n < 25 and n >= 10 and n > 5:
            coins = coins + int(n / 10)
            n = n % 10
        elif n < 25 and n < 10 and n > 1:
            coins = coins + int(n / 5)
            n = n % 5
        else:
            coins +=n
            n = n % 1
    
    print(coins)
            
        
    
def change_owed():
    
    while True:
        n = cs50.get_float("Changed owed: ")
        
        if n > 0:
            break
    
    
    return round(n * 100)
    
main()